app/hooks/use-window-focus.ts

```diff
@@ -15,6 +15,12 @@ export function useWindowFocus() {
   useEffect(() => {
     win.on('focus', focusHandler);
     win.on('blur', blurHandler);
+
+    window.addEventListener('beforeunload', () => {
+      win.removeListener('focus', focusHandler);
+      win.removeListener('blur', blurHandler);
+    });
```

Comment:
What's the use case for this?

Reply:
Prevents errors being thrown in the console/main script